Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -337,7 +308,9 @@ extension KafkaConsumer {
                 source: Producer.Source
             )
             /// The ``KafkaConsumer`` has been closed.
-            case finished
+            ///
+            /// - Parameter client: Client used for handling the connection to the Kafka cluster.
+            case finished(client: KafkaClient)
```

Comment:
That is a bit weird. We should make sure that the end state actually doesn't retain the client so that we can deinit it. Maybe we need a new state here.